src/WithModelQuery.ts

```diff
@@ -31,12 +31,19 @@ export interface IWithModelQueryOptions extends IWithModelQueryProps, IWithLoadi
  */
 export function withModelQuery<P = {}>(options?: IWithModelQueryOptions & P): ComponentEnhancer<P, P> {
```

Comment:
Output prop is a `QueryManager` ?

Reply:
In the student list, we have to do all kinds of acrobatics to get the totalCount off of the `QueryManager`, because the `IWithModelQueryProps` has a `query` property of type `QueryBuilder` on it.

`totalCount={query && (query as any).response && (query as any).response.totalCount}`

Reply:
What you're using in that example is a `QueryManager`.
`withModelQuery` takes `query` as an `IQueryParams | IQueryBuilder`, which is then mapped to a `QueryManager`. The `QueryManager` has on it a `query` which is the `IQueryBuilder` used to make the request, as well as an `response` as an `IQueryResponse`.

While the query is still loading, there is no `response`, so that object is null until we get a response from the API.

Reply:
If I mouse over `{ query, items }` in the spread operator on line 16 of that file, static analysis says that the item coming out of the `withModelQuery` HOC is still an `IQueryParams | IQueryBuilder`.  I don't think it's clear to the consumer of r-d-s-r that they then have to cast it to `any` to get the response `QueryManager` out of it.

Reply:
Sounds like the typing is not setup correctly. I'll look into it.

Reply:
A new interface has been added which should address this

Reply:
💥